micro-rdk/src/esp32/conn/network.rs

```diff
@@ -268,12 +275,36 @@ impl Esp32WifiNetwork {
         &self,
         mut networks: Vec<NetworkSetting>,
     ) -> Result<(), WifiManagerError> {
-        // TODO(RSDK-10184): scan available networks first
-        // attempt to connect only if available
+        self.set_station_mode(NetworkSetting::new("".to_string(), "".to_string(), 0))
```

Comment:
Is this still needed?

Reply:
Will test if needed. I recall some c-based esp-IDF examples setting an empty configuration before starting. 

If not needed will remove.

Reply:
looking a little deeper, I think I'd like to keep this. 

Looking at the EspWifi driver, the behavior of [`start`](https://docs.esp-rs.org/esp-idf-svc/esp_idf_svc/sys/fn.esp_wifi_start.html) is determined by the station mode. So the important thing isn't the empty network settings as much as it is ensuring the[ `ClientConfiguration`](https://docs.esp-rs.org/esp-idf-svc/esp_idf_svc/wifi/struct.ClientConfiguration.htmll) is set appropriately. 

will use Default instead of new here though to make it nicer looking

I haven't tested it, but it's possible it might come up in mixed/AP mode when started if we don't set it appropriately.